### PR TITLE
Post mortem of deployment in 14th December

### PR DESCRIPTION
**Pull request by eduardoj**
Blog post about 14th December's deployment problems.

This PR supersedes #97.

### CODE REVIEW

**Comment by eduardoj** [comment]
All comments in #97 were reviewed. @openSUSE/open-build-service please review.

File: _posts/deployments/2017-12-15-post-mortem-4.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  https://github.com/openSUSE/open-build-service/pull/4200
+
+- ### Remove kiwi image extra version field
+  A simple migration to remove a column in the kiwi images table.
+
+  https://github.com/openSUSE/open-build-service/pull/4209
+
+## What happened?
+
+After yesterday's deployment everything seemed to work properly. We executed the delayed job
```
Comment (bgeuken) [inline _posts/deployments/2017-12-15-post-mortem-4.html]: How about 'Shortly after yesterday's deployment'? I remember it just took a few minutes until you mentioned this and Evan opened a PR with a fix.

File: _posts/deployments/2017-12-15-post-mortem-4.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+we accidentally removed a variable when moving `ProjectLogRotate` to 1 event per job.
+As this is a script only we used, it was not such a problem and it was really fast solved:
+
+https://github.com/openSUSE/open-build-service/pull/4225
+
+And everything seemed to work properly again. Deployment done! Or maybe not...
+
+> Does someone know what all these errors are? 😕
+
+We started receiving the same errors in Errbit. This error was caused
```
Comment (bgeuken) [inline _posts/deployments/2017-12-15-post-mortem-4.html]: I would prefer: "Right after that we noticed the same errors in Errbit, our internal exception tracker."

It makes it a bit clearer how much time passed (not much) after the first issue. And some people might not know Errbit. So let's give them a brief idea.

File: _posts/deployments/2017-12-15-post-mortem-4.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+https://github.com/openSUSE/open-build-service/pull/4227
+
+In one of the newly created classes during the refactor of Event module we had removed one
+of the attributes accepted in the creation of this event. To detect this error we would
+have to try to create an instance of this specific class giving it this attribute.
+
+## How will we do better in the future?
+
+It is clear that it was a bad idea to deploy so many changes at once. The refactor
+of the Event module contained a lot of changes, which we should have deployed alone.
```
Comment (bgeuken) [inline _posts/deployments/2017-12-15-post-mortem-4.html]: Nice:+1:

File: _posts/deployments/2017-12-15-post-mortem-4.html (hunk 1 above)
Comment (eduardoj) [reply _posts/deployments/2017-12-15-post-mortem-4.html]: Applied.

File: _posts/deployments/2017-12-15-post-mortem-4.html (hunk 2 above)
Comment (eduardoj) [reply _posts/deployments/2017-12-15-post-mortem-4.html]: Also applied.

File: _posts/deployments/2017-12-15-post-mortem-4.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+Right after that we noticed the same errors in Errbit, our internal exception tracker. This error was caused
+by the `UpdateNotificationEvents` job, which is run every 17 seconds to create
+events. It was a really easy problem to solve (once we found out what was failing 😜):
+
+https://github.com/openSUSE/open-build-service/pull/4227
+
+In one of the newly created classes during the refactor of Event module we had removed one
+of the attributes accepted in the creation of this event. To detect this error we would
+have to try to create an instance of this specific class giving it this attribute.
```
Comment (bgeuken) [inline _posts/deployments/2017-12-15-post-mortem-4.html]: Let's add a last section here: After 20 minutes of debugging and testing our patch, we managed to solve the issue and Errbit stopped reporting errors.

**Comment by bgeuken** [comment]
@hennevogel We've updated the text a bit and some clearer time when what happened (not a full timeline though). 
Since we don't want to delay this further, we deploy it now. If you want some more changes, let us know and will change it later.

**Pull request merged**